qube/create/templates/portlet/portlet_groovy_osgi/{{ cookiecutter.project_slug }}/src/main/java/life/qbic/portal/portlet/MyPortletUI.java

```diff
@@ -0,0 +1,105 @@
+package life.qbic.portal.portlet;
+
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+
+import com.vaadin.annotations.Theme;
+import com.vaadin.icons.VaadinIcons;
+import com.vaadin.server.VaadinRequest;
+import com.vaadin.server.VaadinService;
+import com.vaadin.shared.ui.ContentMode;
+import com.vaadin.ui.*;
+import com.vaadin.ui.themes.ValoTheme;
+
+import life.qbic.example.Car;
+import life.qbic.example.Garage;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ServiceScope;
+
+
+@Theme("mytheme")
+@SuppressWarnings("serial")
+@Component(
+        service = UI.class,
+        property = {
+        "com.liferay.portlet.display-category=category.sample",  // Widget category
+        "javax.portlet.name=example-osgi-portlet-1.2.0",         // Unique portlet name in Liferay
+        "javax.portlet.display-name=OSGi Portlet",               // Portlet display name
+        "javax.portlet.security-role-ref=power-user,user",       // Default user roles
+        "com.vaadin.osgi.liferay.portlet-ui=true" },             // Is Vaadin UI?
+        scope = ServiceScope.PROTOTYPE)  // New instance created for each distinct request. Mandatory.
+public class MyPortletUI extends UI {
```

Comment:
I think the class with the cookiecutter name was removed by introducing this file. Is this intended?

Reply:
{{ cookiecutter.main_class_prefix }}PortletUI.groovy